### ISSUE TITLE
Build Production - Minified React error #130

### ISSUE THREAD

**Issue by imagine10255** [react-boilerplate/react-boilerplate#2518]
## Description
same error to this:
https://stackoverflow.com/questions/50428339/error-minified-react-error-130

in release, 

develop mode is ok, not error, but only production
also component has this error, no idea

i find this (@babel/plugin-transform-react-constant-element) , and remove after ok!

https://github.com/react-boilerplate/react-boilerplate/blob/master/babel.config.js#L23

thank for this react-boilerplate!

## Steps to reproduce
```
yarn start:production
```

![image](https://user-images.githubusercontent.com/5771353/49840386-22a3bd00-fdee-11e8-8a2f-80ef11bed5d5.png)

## Recommend

same this handle:
https://github.com/facebook/create-react-app/issues/553

## Versions

- React-Boilerplate: 3.7.0
- Node/NPM: node v8.12.0 /  yarn 1.12.3 / npm 6.4.1
- Browser: chrome 71.0.3578.80 x64 (osx)

**Comment by julienben** [comment]
Thanks for reporting!

I'm not sure what to make of this though...

The StackOverflow issue turned out to be a syntax error and the create-react-app issue is from 2016. You can see [here](https://github.com/facebook/create-react-app/blob/5c3f333837e4a296b1d6c7c9844cbb73516d3255/packages/babel-preset-react-app/package.json) that they've gone back to using @babel/plugin-transform-react-constant-elements since then.

Because of this I'm tempted to think that your code is somehow causing this error...? Any chance you can provide more detailed steps to reproduce?

@gretzky You seen this Minified React Error before?

**Comment by imagine10255** [comment]
I will try to keep that commit and upload it to git
thanks

**Comment by gretzky** [comment]
@julienben nope, haven't seen it. I'm also tempted to think there might be some code issue here

**Comment by josebrito** [comment]
Sadly this exact error happened to me too.

Steps to reproduce:

1) install latest `react-boilerplate`

2) edit `app/containers/HomePage/index.js`. Add a div with the `css` prop which is new to `styled-components` v4 and should be supported by `react-boilerplate` since it uses `babel-plugin-styled-components`:

```

**Comment by julienben** [comment]
Will look into it ASAP.

**Comment by gretzky** [comment]
The last time I got this error, it was because the order of babel plugins (`transform-react-constant-elements` and `transform-react-inline-elements`) had to be in a certain order. It sounds weird, but has anyone tried that?

**Comment by josebrito** [comment]
@gretzky I have tried that, but still face the same issue.

**Comment by julienben** [comment]
I added a <div> with the css prop and tested a bunch of variations on the babel config.

At first the build worked when I removed `@babel/plugin-transform-react-inline-elements` entirely.

So I thought it was simply an incompatibility between `@babel/plugin-transform-react-inline-elements` and `babel-plugin-styled-components`.

But now my builds have stopped working, with or without `@babel/plugin-transform-react-inline-elements` (and the `offline-plugin` is disabled) so I'm stumped.

Anybody got any insight on this?

@mxstbr maybe?

**Comment by josebrito** [comment]
No news over this issue?

I performed another test a few days ago, and realized that this would work using the `css` *function* provided by styled-components, but I guess this defeats the whole purpose of the `css` *prop* since it requires an import:

```
import { css } from 'styled-components';

//...

<div css={css`color: red;`}>Hello world</div>
```

By the way, a similar example is provided in the official [blog post](https://medium.com/styled-components/announcing-native-support-for-the-css-prop-in-styled-components-245ca5252feb) regarding the css prop release, but *not* in the [official documentation](https://www.styled-components.com/docs/api#css-prop). 

@mxstbr maybe a mistake here? What should be the recommended usage?

**Comment by josebrito** [comment]
I've provided a [sandbox](https://codesandbox.io/s/github/josebrito/css-prop-minified-error-demo) and a [repo](https://github.com/josebrito/css-prop-minified-error-demo/tree/master/) with a reproducible use case.

**Comment by julienben** [comment]
> I guess this defeats the whole purpose of the css prop since it requires an import

I don't think having to write an import defeats the purpose. From what I understood, the purpose is to not have to write the styled component yourself and let the Babel plugin do that for you.

I have no idea why we see code samples with the tagged template literal though and others without.

Regarding a fix, I personally have no news. Would love some help on this.

(My only other hope is that somehow doing maintenance on the repo will unlock this issue.)

**Comment by josebrito** [comment]
@julienben [it was fixed](https://github.com/styled-components/babel-plugin-styled-components/issues/221) just a few hours ago. Waiting for it to be released!

The sandbox I shared above already reflects the fix.

**Comment by julienben** [comment]
Excellent! Can you let me know when it's released and we'll make sure to upgrade here?

**Comment by josebrito** [comment]
Sure, no problem!

**Comment by josebrito** [comment]
@julienben fix was released under [babel-plugin-styled-components@1.10.3/1.10.4](https://github.com/styled-components/babel-plugin-styled-components/releases/tag/v1.10.3). 

[Demo](https://codesandbox.io/s/github/josebrito/css-prop-minified-error-demo)

All credits to [@probablyup](https://github.com/probablyup).

**Comment by julienben** [comment]
Thanks a lot @josebrito ! The fix will be merged in #2685 and released in v4.1.

(I tested it too and it works great!)
